src/collectors/mesos/mesos.py

```diff
@@ -0,0 +1,68 @@
+# coding=utf-8
+
+"""
+
+Collects metrics from a mesos instance (master or slave).
+
+#### Example Configuration
+
+Typically metrics for the master are exposed on port 5050,
+while metrics for the slaves are exposed on port 5051.
+
+```
+    host = localhost
+    port = 5050
+```
+"""
+
+import diamond.collector
+import json
+import urllib2
+
+
+class MesosCollector(diamond.collector.Collector):
+
+    METRICS_PATH = "metrics/snapshot"
+
+    def get_default_config_help(self):
+        config_help = super(MesosCollector,
+                            self).get_default_config_help()
+        config_help.update({
+            'host': 'Hostname',
+            'port': 'Port',
+            'path': 'Metrics Path'
+        })
+        return config_help
+
+    def get_default_config(self):
+        config = super(MesosCollector, self).get_default_config()
+        config.update({
+            'host': 'localhost',
+            'port': 5050,
```

Comment:
5050 and 5051 are the default values, however these ports are configurable by the user:

http://mesos.apache.org/documentation/latest/configuration/

What we could do, perhaps, is add a flag (is_master) as you've suggested, and pick the default port if not otherwise provided. I think we still need to allow the user to provide it, though, in the event that they have changed it. 

Sound good? Or does that complicate things too much? :)